nextjs-13/src/app/page.test.tsx

```diff
@@ -1,51 +1,107 @@
 import React from 'react';
 
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { redirect } from 'next/navigation';
 import { useSession } from 'next-auth/react';
 
+import requestManager from 'lib/request/manager';
+
 import Home from './page';
 
 jest.mock('next-auth/react');
 jest.mock('next/navigation');
+jest.mock('lib/request/manager');
+
+jest.mock('react-spinners', () => ({
+  ClipLoader: () => <div data-testid="clip-loader">Loading...</div>,
+}));
 
 describe('Home', () => {
-  describe('Session status is "loading', () => {
-    it('renders loading state', () => {
-      useSession.mockReturnValue({ status: 'loading' });
-      render(<Home />);
-      expect(screen.getByText('Loading...')).toBeInTheDocument();
-    });
+  beforeEach(() => {
+    requestManager.mockResolvedValue({ records: [] });
   });
 
   describe('Session status is "unauthenticated', () => {
-    it('redirects to home page', () => {
+    it('redirects to home page', async () => {
       useSession.mockReturnValue({ status: 'unauthenticated' });
+
       render(<Home />);
 
-      expect(redirect).toHaveBeenCalledWith('/auth/sign-in');
+      await waitFor(() => {
+        expect(redirect).toHaveBeenCalledWith('/auth/sign-in');
+      });
     });
   });
 
   describe('Session status is "authenticated', () => {
-    it('renders navigation bar', () => {
+    beforeEach(() => {
       useSession.mockReturnValue({ status: 'authenticated' });
+    });
+
+    it('renders navigation bar', async () => {
       render(<Home />);
 
-      expect(screen.getByText('Newsletter')).toBeInTheDocument();
+      await waitFor(() => {
+        expect(screen.getByText('Newsletter')).toBeInTheDocument();
+      });
     });
 
-    it('renders create newsletter button', () => {
-      useSession.mockReturnValue({ status: 'authenticated' });
+    it('renders title', async () => {
       render(<Home />);
 
-      expect(screen.getByText('Create newsletter')).toBeInTheDocument();
+      await waitFor(() => {
+        expect(screen.getByText('Your Newsletters')).toBeInTheDocument();
+      });
     });
 
-    it('renders a modal to create newsletter when click create newsletter button', async () => {
-      useSession.mockReturnValue({ status: 'authenticated' });
+    describe('Suspense', () => {
```

Comment:
That's a detail of implementation that should not be tests (i.e., it could be removed). In BDD style, focus on describing the behaviors:

```suggestion
    describe('Listing newsletters ', () => {
```

Reply:
@tednguyendev this comment has yet to be addressed

Reply:
Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/110/commits/da187feeeca914879b6d502720a739b9205f3408#diff-4b9ae30a4e34a98e0e0d12af3e4872e7ac9865c68f016c1e6667d4336ce08ce1R58)